english/main.go

```diff
@@ -13,6 +13,8 @@ type WordTokenizer struct {
 }
 
 var reAbbr = regexp.MustCompile(`((?:[\w]\.)+[\w]*\.)`)
+var reLooksLikeEllipsis = regexp.MustCompile(`(?:\.\s?){2,}\.`)
+var reEntities = regexp.MustCompile(`Yahoo!`)
```

Comment:
Yeah, I definitely agree about `reLooksLikeEllipsis`. I think it could be handled in `IsEllipsis`, which currently does not match tokens like `. . .` (but I tried to keep my changes confined to the `english` package).

`reEntities`, IMO, is less of a cheat: I think it's reasonable to have a list of known punctuation-containing entities to avoid marking in `HasSentEndChars` .